tsdb/store.go

```diff
@@ -672,6 +673,26 @@ func (s *Store) Shard(id uint64) *Shard {
 	return sh
 }
 
+// ClearBadShardList will remove all shards from the badShards cache
+// this will allow for lazy loading of bad shards if/when they are no
+// longer in a "bad" state. This method will return any shards that
+// were removed from the cache.
+func (s *Store) ClearBadShardList() map[uint64]error {
+	if s.badShards.shardErrors == nil {
+		badShards := make(map[uint64]error)
+		return badShards
+	}
+	badShards := maps.Clone(s.badShards.shardErrors)
+	clear(s.badShards.shardErrors)
+
+	return badShards
+}
+
+// GetBadShardList is exposed as a method for test purposes
+func (s *Store) GetBadShardList() map[uint64]error {
+	return s.badShards.shardErrors
```

Comment:
This needs to be protected by the mutex here, and the return value cloned for safety.